junit-runtime/src/main/scala/org/scalajs/junit/JUnitExecuteTest.scala

```diff
@@ -0,0 +1,227 @@
+                testAnnotation, timeInSeconds)
+
+            afterMethods match {
+              case Success (_) => // Do nothing
+              case Failure(exception) => ???
+            }
+
+          case Failure(exception) =>
+            executedWithExceptions(packageName, className, method.name,
+                testAnnotation, timeInSeconds, exception)
+        }
+
+      case Failure(ex: AssumptionViolatedException[_]) =>
+        logFormattedInfo(packageName, className, method.name, "started")
+        logAssertionWarning(packageName, className, method.name, ex,
+            timeInSeconds)
+
+      case Failure(ex: internal.AssumptionViolatedException) =>
+        logFormattedInfo(packageName, className, method.name, "started")
+        logAssertionWarning(packageName, className, method.name, ex,
+            timeInSeconds)
+
+      case Failure(ex) => throw ex
+    }
+
+    //  if (testAnnotation.timeout != 0 && testAnnotation.timeout <= time) {
+    //    richLogger.warn ("Timeout: took " + timeInSeconds + " sec, expected " +
+    //      (testAnnotation.timeout.toDouble / 1000) + " sec")
+    //  }
+
+  }
+
+  private[this] def ignoreTest(packageName: String, className: String,
+      methodName: String) = {
+    logFormattedInfo(packageName, className, methodName, "ignored")
+    runner.taskSkipped()
+    runner.taskRegisterTotal()
+  }
+
+  private[this] def executedWithoutExceptions(packageName: String,
+      className: String, methodName: String, testAnnotation: org.junit.Test,
+      timeInSeconds: Double) = {
+    if (testAnnotation.expected == classOf[org.junit.Test.None]) {
+      if (runner.runSettings.verbose)
+        logFormattedInfo(packageName, className, methodName, "started")
+      runner.taskPassed()
+    } else {
+      val msg = {
+        s"failed: Expected exception: ${testAnnotation.expected} " +
+        s"took $timeInSeconds sec"
+      }
+      logFormattedError(packageName, className, methodName, msg)
+      runner.taskFailed()
+    }
+    runner.taskRegisterTotal()
+  }
+
+  private[this] def executedWithExceptions(packageName: String,
+      className: String, methodName: String, testAnnotation: org.junit.Test,
+      timeInSeconds: Double, ex: Throwable) = {
+    if (classOf[AssumptionViolatedException[_]].isInstance(ex) ||
+        classOf[internal.AssumptionViolatedException].isInstance(ex)) {
+      logFormattedInfo(packageName, className, methodName, "started")
+      logAssertionWarning(packageName, className, methodName, ex, timeInSeconds)
+      runner.taskSkipped()
+    } else if (testAnnotation.expected.isInstance(ex)) {
+      if (runner.runSettings.verbose)
+        logFormattedInfo(packageName, className, methodName, "started")
+      runner.taskPassed()
+    } else if (testAnnotation.expected == classOf[org.junit.Test.None]) {
+      val failedMsg = new StringBuilder
+      failedMsg ++= "failed: "
+      if (ex.getClass == classOf[AssertionError] &&
```

Comment:
Must this actually be `getClass ==`?
Shouldn't it be `ex.isInstanceOf[AssertionError]`?